tianshou/data/collector.py

```diff
@@ -245,26 +258,27 @@ def collect(self,
             # convert None to Batch(), since None is reserved for 0-init
             if state is None:
                 state = Batch()
-            self.data.update(state=state, policy=result.get('policy', Batch()))
+            # since result is a Batch, it can bypass the type check here
+            self.data.__dict__['state'] = state
```

Comment:
I'm not a huge fan of bypassing stuff. It makes things harder to maintain. Indeed, it is quite usual to change the internal behaviour without changing its external API. So the performance improvement is not significant, I would rather stick to the official API. But in this case it is not a serious issue.

Reply:
Hmm, I'll compare those two versions

Reply:
Is it true when ``state`` is large (e.g. in RNN), too?